File: src/main/java/no/bekk/java/exercises/Ex4_LambdaOnParallelCollections.java
package no.bekk.java.exercises;

import no.bekk.java.model.Team;

import java.util.List;
import java.util.function.Predicate;

import static java.util.stream.Collectors.toList;
import static no.bekk.java.util.Utils.implementThis;

public class Ex4_LambdaOnParallelCollections {

	static List<Team> teamsWithValueHigherThan(Double value, List<Team> teams) {
		return implementThis();
	}

	private static <T> Predicate<T> slowPredicate(Predicate<T> predicate) {
		return x -> {
			try{
				Thread.sleep(100);
			} catch(InterruptedException e) {
				e.printStackTrace();
			}
			return predicate.test(x);
		};
	}
}

File: src/main/scala/no/bekk/scala/exercises/Ex4_LambdaOnParallelCollections.scala
package no.bekk.scala.exercises

import no.bekk.scala.model.Team

object Ex4_LambdaOnParallelCollections {

	def teamsWithValueHigherThan(value: Double, teams: List[Team]): List[Team] = ???

	private def slowPredicate[A](predicate: A => Boolean): A => Boolean = x => {
		try {
			Thread.sleep(100)
		} catch {
			case e: InterruptedException => e.printStackTrace()
		}
		predicate(x)
	}
}
